ios/RNSScreen.mm

```diff
@@ -964,6 +1005,83 @@ - (void)viewDidLayoutSubviews
   }
 }
 
+- (void)dismissViewControllerAnimated:(BOOL)flag completion:(void (^)())completion
+{
+  [super dismissViewControllerAnimated:flag completion:completion];
+  [self calculateHeaderHeightIsModal:self.screenView.isPresentedAsNativeModal];
+}
+
+// Checks whether this screen has any child view controllers of type RNScreensNavigationController.
+// Useful for checking if this screen has nested stack or is displayed at the top.
+- (BOOL)hasAnyChildNavigators
+{
+  for (int i = 0; i < [self.childViewControllers count]; i++) {
+    if ([self.childViewControllers[i] isKindOfClass:[RNScreensNavigationController class]]) {
+      return true;
+    }
+  }
+
+  return false;
+}
+
+- (CGFloat)getCalculatedHeaderHeightIsModal:(BOOL)isModal
+{
+  CGFloat navbarHeight = self.navigationController.navigationBar.frame.size.height;
+
+  // In case where screen is a modal, we want to calculate just its childViewController's height
+  if (isModal && self.childViewControllers.count > 0 &&
+      [self.childViewControllers[0] isKindOfClass:UINavigationController.class]) {
+    UINavigationController *childNavCtr = self.childViewControllers[0];
+    navbarHeight = childNavCtr.navigationBar.frame.size.height;
+  }
+
+  return navbarHeight;
+}
+
+- (CGSize)getCalculatedStatusBarHeightIsModal:(BOOL)isModal
+{
+#if !TARGET_OS_TV
+  BOOL isDraggableModal = isModal && ![self.screenView isFullscreenModal];
+  BOOL isDraggableModalWithChildViewCtr =
+      isDraggableModal && self.childViewControllers.count > 0 && self.childViewControllers[0] != nil;
+
+  // When modal is floating (we can grab its header), we don't want to calculate status bar in it.
+  // Thus, we return '0' as a height of status bar.
+  if (isDraggableModalWithChildViewCtr || self.screenView.isTransparentModal) {
+    return CGSizeMake(0, 0);
+  }
+
+  CGSize fallbackStatusBarSize = [[UIApplication sharedApplication] statusBarFrame].size;
+
+  if (@available(iOS 13.0, *)) {
+    CGSize primaryStatusBarSize = self.view.window.windowScene.statusBarManager.statusBarFrame.size;
+    if (primaryStatusBarSize.height == 0 || primaryStatusBarSize.width == 0)
+      return fallbackStatusBarSize;
+
+    return primaryStatusBarSize;
+  } else {
```

Comment:
This code should be put inside ifdefs, as for someone on really old xcode it would fail to build. See [here](https://github.com/software-mansion/react-native-screens/blob/d40dc028a6e69a9aefbe56f6098fa9f517fc8190/ios/RNSScreen.mm#L120) for example.

Reply:
Sure! I was thinking why we're putting those ifdefs 🤔

Reply:
FYI: The ```if (@available(iOS ... ))``` check is evaluated at runtime. When someone uses older xcode version without support for sdk >= iOS 13 the code inside the block could fail to build because of missing symbols at compile time.

Reply:
Introduced in https://github.com/software-mansion/react-native-screens/pull/1802/commits/1f5d586aff8373e79fa535af38756bd4c6da8e68.